### PR TITLE
Remove `is_deterministic` from system.functions

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Remove `is_deterministic` field from the `system.functions` table.

The implementation was wrong. The proper implementation would be: a properties structure in the factory, similar to aggregate functions. A normal code path cannot throw exceptions, and we have a test for that.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: tests/queries/0_stateless/02815_no_throw_in_simple_queries.sh [hunk 1]
```diff
@@ -45,9 +45,4 @@ expect eof
 EOF
 }
 
-run "$CLICKHOUSE_LOCAL --disable_suggestion"
-# Suggestions are off because the suggestion feature initializes itself by reading all available function
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/02815_no_throw_in_simple_queries.sh:49]: The problem was here.
Bottom line: tests exist for a reason.

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 855f03d1d3c5b34d3de1b0919b9f7975d64b3cb8 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66630/855f03d1d3c5b34d3de1b0919b9f7975d64b3cb8/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/66630/855f03d1d3c5b34d3de1b0919b9f7975d64b3cb8/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

**Comment by alexey-milovidov** [comment]
Needs #66694

**Review by davenger — APPROVED** [review]

**Comment by rschu1ze** [comment]
Re-implementing the removed feature here: https://github.com/ClickHouse/ClickHouse/pull/66697

**Pull request merged**